lib/evmone/instructions.hpp

```diff
@@ -783,6 +783,57 @@ inline void gas(StackTop stack, ExecutionState& state) noexcept
     stack.push(state.gas_left);
 }
 
+namespace
+{
+
+template <size_t Len>
+uint64_t load(code_iterator pos) noexcept
+{
+    static_assert(Len > 4 && Len < 8);
+
+    uint64_t d;  // NOLINT(cppcoreguidelines-init-variables)
+    __builtin_memcpy(&d, pos, sizeof(uint64_t));
+    d = __builtin_bswap64(d);
+    d = d >> (64 - 8 * Len);
+    return d;
+}
+
+template <>
+uint64_t load<1>(code_iterator pos) noexcept
+{
+    return pos[0];
+}
+
+template <>
+uint64_t load<2>(code_iterator pos) noexcept
+{
+    uint16_t d;  // NOLINT(cppcoreguidelines-init-variables)
+    __builtin_memcpy(&d, pos, sizeof(d));
+    d = __builtin_bswap16(d);
+    return d;
```

Comment:
Here we CANNOT use `intx::be::unsafe::load<uint16_t>` because it does not work for types smaller than `uint64_t`.